### PR TITLE
Bug/event for tilesource open

### PR DESCRIPTION
**Pull request by nein09**
Moved from #95.

### CODE REVIEW

File: example/index.html [hunk 1]
```diff
@@ -105,6 +105,10 @@
                  viewer.next();
              });
 
+            // Debug/example code: Listen for tile source requests and loads
+            viewer.on('detail-tile-source-requested', function(e) { console.log('detail tile source requested', e.detail)});
+            viewer.on('detail-tile-source-opened', function(e) { console.log('detail tile source opened', e.detail)});
+
```
Comment (aeschylus) [inline example/index.html:111]: Yep. This is the kind of pattern I was thinking of. Good to pass in the global dispatcher (below) as well. Thanks for setting this up. I think I can set up an example page for the layering interface using this kind of API (will use some imaginary event names for now).

**Comment by iangilman** [comment]
@aeschylus I know this task isn't complete, but maybe this much is already a good chunk to merge in?

**Comment by nein09** [comment]
@iangilman @aeschylus I'm basing the next chunk of work (for issue 40) off of this branch anyway, FYI.

**Comment by aeschylus** [comment]
:+1:

**Pull request merged**